File: p1-easy/HarkiratAndHisGirlfriends_sol.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int t; cin >> t; // o. of tc
    while(t--){
        int n,q; cin >> n >> q; // Number of girlfriends and number of queries

        map<string,int> mp;// // HashMap to store gifts sum for each girlfriend

        for (int i = 0; i < n; ++i)
        {
            string gf_name; cin >> gf_name;
            int no_of_gifts; cin >> no_of_gifts;
            int sum=0;

            for (int j = 0; j < no_of_gifts; ++j)
            {
                int price; cin >> price;
                sum+=price;
            }
            mp[gf_name]=sum;
        }

        while(q--){
            string query_gf; cin >> query_gf;
            cout << mp[query_gf] << endl;
        }
    }
    return 0;
}

File: p1-easy/HarkiratAndHisGirlfriends_sol.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt(); // Number of test cases
        
        while (t-- > 0) {
            int n = sc.nextInt(); // Number of girlfriends
            int q = sc.nextInt(); // Number of queries
            
            Map<String, Integer> mp = new HashMap<>(); // HashMap to store gifts sum for each girlfriend
            
            for (int i = 0; i < n; ++i) {
                String gf_name = sc.next();
                int no_of_gifts = sc.nextInt();
                int sum = 0;
                
                for (int j = 0; j < no_of_gifts; ++j) {
                    int price = sc.nextInt();
                    sum += price;
                }
                mp.put(gf_name, sum);
            }
            
            while (q-- > 0) {
                String query_gf = sc.next();
                System.out.println(mp.getOrDefault(query_gf, 0));
            }
        }
        sc.close();
    }
}

